cpp/src/arrow/status.cc

```diff
@@ -131,6 +131,17 @@ void Status::Abort(const std::string& message) const {
   std::abort();
 }
 
+void Status::Warn() const { Warn(std::string()); }
+
+void Status::Warn(const std::string& message) const {
+  ARROW_LOG(WARNING) << "-- Arrow Warning --";
+  if (!message.empty()) {
+    ARROW_LOG(WARNING) << message;
+  }
+
+  ARROW_LOG(WARNING) << ToString() << "\n";
```

Comment:
FWIW, it might be better to consolidate the message and status into a single log so that concurrent logs don't get mixed up.

```suggestion
  ARROW_LOG(WARNING) << "-- Arrow Warning --";
  if (!message.empty()) {
    ARROW_LOG(WARNING) << message;
  }

  ARROW_LOG(WARNING) << ToString();
```

Reply:
I would honestly just make it

```suggestion
  ARROW_LOG(WARNING) << "-- Arrow Warning --";
  ARROW_LOG(WARNING) << message << ToString();
```

Reply:
That way even the original status messages will keep working

Reply:
Here's an update to what the logging looks like. I think left alignment leaves something to be desired:
```
/Users/octalene/code/arrow-dev/cpp/src/arrow/status.cc:137: -- Arrow Warning --
Test warning message
Invalid: warn_summary
```

Reply:
Why not just put it all on one line? `WARNING: Warning message verbatim: Invalid: warn_summary`

Reply:
will do!